### ISSUE TITLE
Simplified and more efficient Binarize implementation

### ISSUE THREAD

**Issue by glencoe** [es-ude/elastic-ai.creator#253]
This implementation

```
class Binarize(torch.nn.Module):
    def forward(self, x):
        with torch.no_grad():
            bx = torch.where(torch.gt(x, 0), -1, 1)
            dy = bx - x
        return x + dy
```

would applies a straight through estimator without the need of using custom autograd functions.
This at least on my cpu, drastically improves inference speed during training. And it is shorter than the current version.
I suggest we replace the current implementation with the above.

**Comment by glencoe** [comment]
after reconsideration... not sure how to clip gradients with this implementation.

**Comment by julianhoever** [comment]
@glencoe Why do we need to clip the gradients if we want to apply a STE? I thought that this just passes the gradients through without modifications.
